controller/controller.go

```diff
@@ -994,7 +994,10 @@ func (ctrl *ProvisionController) syncClaim(obj interface{}) (ProvisioningState,
 		return ProvisioningFinished, fmt.Errorf("expected claim but got %+v", obj)
 	}
 
-	if ctrl.shouldProvision(claim) {
+	should, err := ctrl.shouldProvision(claim)
```

Comment:
Add some unit tests for the new conditions?

Reply:
added 2: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/38/commits/d8b10d67987b54a2fe6f247b58781d628c8d1a7a#diff-33074880ee4a8617ea1496028ff0aa30R650